blockchain/validation.go

```diff
@@ -69,8 +121,51 @@ func (bc *BlockChain) ValidTransaction(t *Transaction) (bool, TransactionCode) {
 	return true, ValidTransaction
 }
 
+// ValidCloudBase returns true if a transaction is a valid CloudBase transaction
+// and false otherwise
+func ValidCloudBase(t *Transaction) (bool, CloudBaseTransactionCode) {
```

Comment:
Yeah it's fine to take the address and block reward rather than the block. We just need to verify those cases somehow